src/site/content/en/blog/read-files/index.md

```diff
@@ -0,0 +1,253 @@
+* [Read a file's content](#read)
+* [Slice a file's content](#slice)
+* [Monitor the progress of a file read](#monitor)
+
+## Browser compatibility {: #compatibility }
+
+Detailed browser compatibility data is listed at the bottom of each section.
+If you see the usual [feature detection design pattern][detection] in a code sample
+it means that it's unknown whether the API is 100% supported across all browsers
+at the time of writing.
+
+## Prerequisites {: #prerequisites }
+
+This guide assumes you're familiar with:
+
+* [Beginner web development concepts](https://developer.mozilla.org/docs/Learn)
+* [Events](https://developer.mozilla.org/docs/Learn/JavaScript/Building_blocks/Events)
+* [Feature detection][detection]
+* Modern JavaScript syntax like
+  [arrow functions](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions),
+  [shorthand object declarations](https://alligator.io/js/object-property-shorthand-es6/),
+  [`const`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/const), and
+  [template literals](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Template_literals)
+
+## Select files with an `<input type="file">` element {: #input }
+
+```html
+<!-- The `multiple` attribute lets users select multiple files. -->
+<input type="file" id="input" multiple/>
+<script>
+  if (window.FileList && window.File) {
+    document.getElementById('input').addEventListener('change', event => {
+      const files = event.target.files; // This is a `FileList` object.
+      // `forEach()` won't work unless you convert `files` to an `Array`.
+      for (let i = 0; i < files.length; i++) {
+        const file = files[i]; // This is a `File` object.
+        const name = file.name ? file.name : 'NOT SUPPORTED';
+        const type = file.type ? file.type : 'NOT SUPPORTED';
+        console.log({name, type});
+      }
+    }); 
+  }
+</script>
+```
+
+This example lets a user select multiple files using their operating system's built-in
+file selection UI and then logs out the name and file type of each selected file to the Console.
+
+[`<input type="file">`][input] is the recommended way to enable users to select files because
+browser vendors do a lot of work to make sure that their built-in elements are accessible.
+Some browsers also treat `<input type="file">` elements as automatic drag-and-drop targets.
+After the user clicks the `<input type="file">` element the operating system's built-in
+file selection UI appears. Add the `multiple` attribute to your `<input type="file">` element
+to instruct the operating system to allow the user to select multiple files. When the
+user finishes selecting a file or files the browser fires the `<input type="file">` element's
+`change` event. You can access the list of files from `event.target.files`, which is a [`FileList`][filelist]
+object, not an `Array`. If you want to iterate over `files` with `forEach()` you'll need to convert
```

Comment:
Is `for ... of` the thing to recommend these days? I prefer it to `forEach()`.

Reply:
Yeah I vaguely recall people not liking `forEach()` so I'll just remove that outright. The code samples are now using `for ... of` BTW... @samdutton gave me the same feedback in his Doc.